src/config.js

```diff
@@ -1,44 +1,69 @@
 'use strict'
 
 const Key = require('interface-datastore').Key
+const queue = require('async/queue')
+const waterfall = require('async/waterfall')
+const _get = require('lodash.get')
+const _set = require('lodash.set')
+const _has = require('lodash.has')
 
 const configKey = new Key('config')
 
 module.exports = (store) => {
-  return {
+  const setQueue = queue(_doSet, 1)
+
+  const configStore = {
     /**
      * Get the current configuration from the repo.
      *
+     * @param {String} key - the config key to get
      * @param {function(Error, Object)} callback
      * @returns {void}
      */
-    get (callback) {
-      store.get(configKey, (err, value) => {
+    get (key, callback) {
+      if (typeof key === 'function') {
+        callback = key
+        key = undefined
+      }
+      store.get(configKey, (err, encodedValue) => {
         if (err) {
           return callback(err)
         }
 
         let config
         try {
-          config = JSON.parse(value.toString())
+          config = JSON.parse(encodedValue.toString())
         } catch (err) {
           return callback(err)
         }
-        callback(null, config)
+        if (key !== undefined && !_has(config, key)) {
+          callback(new Error('Key does not exist in config'))
```

Comment:
would be nice to say which key doesn't exist, i.e.`Key "foo" does not exist in config`